ocaml/xapi/message_forwarding.ml

```diff
@@ -2628,6 +2635,33 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 			Xapi_pool_patch.destroy ~__context ~self
 	end
 
+	module Pool_update = struct
+		let introduce ~__context ~vdi =
+			info "Pool_update.introduce: vdi = '%s'" (vdi_uuid ~__context vdi);
+			let local_fn = Local.Pool_update.introduce ~vdi in
+			let host = Helpers.get_master __context in
+			do_op_on ~local_fn ~__context ~host
```

Comment:
Since this is always executed on the master, it does not need to be forwarded, and you don't need `do_op_on`.